### DISCUSSION TITLE
How to install on Debian / nginx

### DISCUSSION THREAD

**Discussion by adevoss** [searxng/searxng#2653] [Q&A] [answered]
Hi,

How do I configure nginx? The example is not a valid nginx configuration file.
When I add
```
server {
        listen 80;
```
at the top and
` }`
at the bottom 
nginx starts. To be sure I disabled the default page.

For installation I used the script (all default options).
After executing the script I tried to configure nginx. Is that the last step?

Now I get a `502 Bad Gateway` at http://ipadress/searxng on local lan.

**Comment by mzch** [comment, marked as answer]
Sample nginx conf when SearXNG listens on port 8080:

```nginx
server {
	listen 80;
	listen [::]:80;
	server_name searxng.example.com;
	if ($host = searxng.example.com ) {
		return 301 https://$host$request_uri;
	}
	return 404;
}

server {

	listen 443 ssl; # managed by Certbot
	listen [::]:443 ssl; # managed by Certbot
	http2 on;

	server_name searxng.example.com;

	ssl_certificate /etc/letsencrypt/live/searxng.example.com/fullchain.pem; # managed by Certbot
	ssl_certificate_key /etc/letsencrypt/live/searxng.example.com/privkey.pem; # managed by Certbot
	include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
	ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

	location / {
		proxy_pass http://127.0.0.1:8080;
		proxy_set_header Host $http_host;
		proxy_set_header X-Real-IP $remote_addr;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header X-Forwarded-Proto https;
		proxy_set_header X-Nginx-Proxy true;
	}

	access_log /var/log/nginx/searxng.example.coms.log;
	error_log /var/log/nginx/searxng.example.com.error.log;
}
```

**Reply by adevoss** [reply]
The default install sets port 888. After changing the website showed up.
Thanks.

**Reply by BaconWillRule** [reply]
I have the same issue with the http error 502 showing up but I have used the docker image, can anyone help?

**Reply by return42** [reply]
Docker uses Caddy / this discussion is about NGINX .. please have a research in https://github.com/searxng/searxng-docker
